src/Codeception/Specify.php

```diff
@@ -1,224 +1,70 @@
-     */
-    public function getCurrentSpecifyTest()
-    {
-        return $this->currentSpecifyTest;
-    }
-
-    public function should($specification, \Closure $callable = null, $params = [])
-    {
-        $this->specify("should " . $specification, $callable, $params);
-    }
-
-    public function it($specification, \Closure $callable = null, $params = [])
-    {
-        $this->specify($specification, $callable, $params);
-    }
-
-    public function describe($specification, \Closure $callable = null)
-    {
-        $this->specify($specification, $callable);
-    }
-
-    public function specify($specification, \Closure $callable = null, $params = [])
-    {
-        if (!$callable) {
-            return;
-        }
-
-        /** @var $this TestCase  **/
-        if (!$this->copier) {
-            $this->copier = new \DeepCopy\DeepCopy();
-            $this->copier->skipUncloneable();
-        }
-
-        $properties = $this->getSpecifyObjectProperties();
-
-        // prepare for execution
-        $examples = $this->getSpecifyExamples($params);
-        $showExamplesIndex = $examples !== [[]];
-
-        $specifyName = $this->specifyName;
-        $this->specifyName .= ' ' . $specification;
-
-        foreach ($examples as $idx => $example) {
-            $test = new SpecifyTest($callable->bindTo($this));
-            $this->currentSpecifyTest = $test;
-            $test->setName($this->getName() . ' |' . $this->specifyName);
-            $test->setExample($example);
-            if ($showExamplesIndex) {
-                $test->setName($this->getName() . ' |' . $this->specifyName . ' # example ' . $idx);
-            }
-
-            // copy current object properties
-            $this->specifyCloneProperties($properties);
-
-            if (!empty($this->beforeSpecify) && is_array($this->beforeSpecify)) {
-                foreach ($this->beforeSpecify as $closure) {
-                    if ($closure instanceof \Closure) $closure->__invoke();
-                }
-            }
-
-            $test->run($this->getTestResultObject());
-            $this->specifyCheckMockObjects();
-
-            // restore object properties
-            $this->specifyRestoreProperties($properties);
-
-            if (!empty($this->afterSpecify) && is_array($this->afterSpecify)) {
-                foreach ($this->afterSpecify as $closure) {
-                    if ($closure instanceof \Closure) $closure->__invoke();
-                }
-            }
-        }
-
-        // revert specify name
-        $this->specifyName = $specifyName;
-    }
-
-    /**
-     * @param $params
-     * @return array
-     * @throws \RuntimeException
-     */
-    private function getSpecifyExamples($params)
-    {
-        if (isset($params['examples'])) {
-            if (!is_array($params['examples'])) throw new \RuntimeException("Examples should be an array");
-            return $params['examples'];
-        }
-        return [[]];
+    use SpecifyBoostrap {
```

Comment:
What about this naming:

`SpecifyHooks`

Before and After are hooks, we can name it like this